src/main/java/com/onarandombox/MultiverseCore/utils/WorldManager.java

```diff
@@ -96,127 +97,203 @@ public boolean accept(File file, String s) {
+        private String newName;
+        private File oldWorldFile;
+        private File newWorldFile;
+        private MultiverseWorld oldWorld;
+
+        public CloneHelper(String oldName, String newName) {
+            this.oldName = oldName;
+            this.newName = newName;
+        }
 
-    /**
-     * {@inheritDoc}
-     */
-    @Override
-    public boolean cloneWorld(String oldName, String newName) {
-        // Make sure we already know about the old world and that we don't
-        // already know about the new world.
-        if (!this.worldsFromTheConfig.containsKey(oldName)) {
-            for (Map.Entry<String, WorldProperties> entry : this.worldsFromTheConfig.entrySet()) {
-                if (oldName.equals(entry.getValue().getAlias())) {
-                    oldName = entry.getKey();
-                    break;
-                }
+        /**
+         * The clone operation.
+         *
+         * @return True if clone is successful, false otherwise.
+         */
+        public boolean run() {
+            if (!validCloneParameters()) {
+                Logging.warning("Invalid clone parameters");
+                return false;
             }
-            if (!this.worldsFromTheConfig.containsKey(oldName)) {
-                Logging.warning("Old world '%s' does not exist", oldName);
+
+            initWorldFiles();
+            if (newWorldFile.exists()) {
+                Logging.warning("Folder for new world '%s' already exists", newName);
                 return false;
             }
-        }
-        if (this.isMVWorld(newName)) {
-            Logging.warning("New world '%s' already exists", newName);
-            return false;
-        }
 
-        // Check for valid world name
-        if (!(isValidWorldName(oldName) && isValidWorldName(newName))) {
-            return false;
+            oldWorld = getMVWorld(oldName);
+            if (!isOldWorldLoadable()) {
+                Logging.severe("Old world '%s' is not loadable", oldName);
+                return false;
+            }
+
+            boolean wasAutoSave = manuallySaveWorld();
+
+            if (!copyWorldFolder()) {
+                return false;
+            }
+
+            revertBackAutoSave(wasAutoSave);
+
+            return initNewCloneWorld();
         }
 
-        final File oldWorldFile = new File(this.plugin.getServer().getWorldContainer(), oldName);
-        final File newWorldFile = new File(this.plugin.getServer().getWorldContainer(), newName);
-        final List<String> ignoreFiles = new ArrayList<>(Arrays.asList("session.lock", "uid.dat"));
+        /**
+         * Ensure that world with oldName can be clone to a world called newName.
+         *
+         * @return True if parameters are valid, otherwise false.
+         */
+        private boolean validCloneParameters() {
+            String originalOldName = oldName;
+            oldName = isWorldFromConfig(oldName) ? oldName : getWorldByAliasFromConfig(oldName);
```

Comment:
Do not try to change the `oldName` variable to the same if `isWorldFromConfig` is true to avoid misunderstands.
```suggestion
           if (!isWorldFromConfig(oldName)) {
               oldName = getWorldByAliasFromConfig(oldName);
           }
```